docs/src/literate-tutorials/incompressible_elasticity.jl

```diff
@@ -116,37 +117,29 @@ end;
 # The element routine integrates the local stiffness and force vector for all elements.
 # Since the problem results in a symmetric matrix we choose to only assemble the lower part,
 # and then symmetrize it after the loop over the quadrature points.
-function assemble_up!(Ke, fe, cell, cellvalues_u, cellvalues_p, facevalues_u, grid, mp, ɛdev, t)
-
-    n_basefuncs_u = getnbasefunctions(cellvalues_u)
-    n_basefuncs_p = getnbasefunctions(cellvalues_p)
-    u▄, p▄ = 1, 2
-    reinit!(cellvalues_u, cell)
-    reinit!(cellvalues_p, cell)
-
+function assemble_up!(Ke, fe, cell, cellvalues, facevalues_u, grid, mp, ɛdev, t, dofrange_u, dofrange_p)
+    reinit!(cellvalues, cell)
     ## We only assemble lower half triangle of the stiffness matrix and then symmetrize it.
-    for q_point in 1:getnquadpoints(cellvalues_u)
-        for i in 1:n_basefuncs_u
-            ɛdev[i] = dev(symmetric(shape_gradient(cellvalues_u, q_point, i)))
+    for q_point in 1:getnquadpoints(cellvalues)
+        for i in keys(dofrange_u)
+            ɛdev[i] = dev(symmetric(shape_gradient(cellvalues[:u], q_point, i)))
         end
-        dΩ = getdetJdV(cellvalues_u, q_point)
-        for i in 1:n_basefuncs_u
-            divδu = shape_divergence(cellvalues_u, q_point, i)
-            δu = shape_value(cellvalues_u, q_point, i)
-            for j in 1:i
-                Ke[BlockIndex((u▄, u▄), (i, j))] += 2 * mp.G * ɛdev[i] ⊡ ɛdev[j] * dΩ
+        dΩ = getdetJdV(cellvalues, q_point)
+        for (iᵤ, Iᵤ) in pairs(dofrange_u)
+            for (jᵤ, Jᵤ) in pairs(dofrange_u[1:iᵤ])
```

Comment:
In my benchmarks, using block indices for the local matrices are always slow. 
Furthermore, it means that if you change the order the fields are added to your dofhandler, you'll silently get wrong results as opposed to using the dofrange directly.

Reply:
Note that the new code will still give you the wrong result in this case.

Reply:
Did I miss an assumption of the ordering of the fields?

Reply:
Note from Slack communication: Yes, we assume u then p ordering when symmetrizing. 
This could potentially be moved to a "performance optimization" how-to to keep this first multifield tutorial as simple as possible.